TimeTracer/background/background.js

```diff
@@ -53,19 +56,72 @@ async function updateActiveUrlSession(newActiveUrl, stopTracking) {
 function tabEnterOrChangeAction(activeUrl, logMsg) {
   activeUrl = cleanUrl(activeUrl);
 
-  // check and redirect
+  // TODO: check and redirect
   //checkBlockedUrls(activeUrl);
 
   updateActiveUrlSession(activeUrl, false);
   __logger__(`${logMsg} ${activeUrl}`, true);
 }
 
+/**
+ * Asynchronously checks and updates the accuracy of the time spent on a website by
+ *  comparing time stamps at set time periods to see if the users device is active or
+ *  asleep.
+ *
+ * This function acts as a wrapper that allows the await key word.
+ *
+ * @returns {Promise<void>} A Promise that resolves when the operation is complete.
+ */
+async function checkIntervalWraper() {
+  let urlData = await getSiteObjData();
+  urlData = checkInterval(urlData, TME_CHECK_INTERVAL_MINUTES);
+  setSiteObjData(urlData);
+}
+
+/**
+ * Creates a repeating alarm using the Chrome Alarms API.
+ * This alarm will fire periodically at the specified interval.
+ *
+ * @param {string} alarmName - The unique name for the alarm.
+ * @param {number} alarmInterval - The interval in minutes after which the alarm should repeat.
+ */
+function createRepeatingAlarm(alarmName, alarmInterval) {
+  chrome.alarms.create(alarmName, {
+    periodInMinutes: alarmInterval,
+  });
+  __logger__(
+    `Alarm '${alarmName}' created to fire every ${alarmInterval} minutes.`
+  );
+}
+
 // ===================================================== \\
 // ===================================================== \\
 //              Chromium API Event Listeners             \\
 // ===================================================== \\
 // ===================================================== \\
 
+// ===================================================== \\
+//                         Alarm                         \\
+// ===================================================== \\
+
+// Listen for the alarm event
+chrome.alarms.onAlarm.addListener((alarm) => {
+  if (alarm.name === TIME_CHECK_ALARM) {
+    __logger__('Alarm fired! -----------------------------------> Alarm');
```

Comment:
I wouldn't log this. If you need logging probably put it in the checkIntervalWrapper.